services/net/syndication/SyndicationAction.cs

```diff
@@ -228,6 +229,32 @@ private static SourceContent CreateSourceContent(string source, SyndicationItem
         };
     }
 
+    /// <summary>
+    /// Handle invalid encoding characters before sending to Kafka.
+    /// </summary>
+    /// <param name="item"></param>
+    /// <returns>updated title and summary</returns>
+    private (string title, string summary) HandleInvalidEncoding(SyndicationItem item)
+    {
+        var title = item.Title.Text;
+        var summary = item.Summary.Text;
+        if (!string.IsNullOrEmpty(this.Options.InvalidEncodings) && this.Options.EncodingSets != null)
+        {
+            foreach (var encodingSet in this.Options.EncodingSets)
+            {
+                var keyValue = encodingSet.Split(":_", StringSplitOptions.RemoveEmptyEntries);
+                if (keyValue?.Length == 2)
+                {
+                    var oldValue = keyValue[0];
+                    var newValue = keyValue[1];
+                    if (title.Contains(oldValue)) title = title.Replace(oldValue, newValue);
```

Comment:
No need to change, but this is only beneficial if the `oldValue` is discovered early in the string.  Essentially you're iterating over the string twice instead of one time by only calling `title.Replace(oldValue, newValue)`.